bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/services/ast/AstNode.java

```diff
@@ -0,0 +1,167 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Advantest Europe GmbH and others.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * Dietrich Travkin (Solunar GmbH) - Initial implementation
+ *******************************************************************************/
+package org.eclipse.cdt.lsp.services.ast;
+
+import java.util.Arrays;
+
+import org.eclipse.cdt.lsp.services.ClangdLanguageServer;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.jsonrpc.validation.NonNull;
+import org.eclipse.lsp4j.util.Preconditions;
+import org.eclipse.lsp4j.util.ToStringBuilder;
+
+/**
+ * Return type for the <em>textDocument/ast</em> request.
+ * This class was generated by the <em>org.eclipse.lsp4j.generator</em> bundle
+ * using xtend (see {@link org.eclipse.lsp4j.generator.JsonRpcData JsonRpcData} and
+ * the <a href="https://github.com/eclipse-lsp4j/lsp4j/blob/main/documentation/jsonrpc.md">documentation</a>).
+ *
+ * @see {@link ClangdLanguageServer#getAst(AstParams)}
+ */
+public class AstNode {
```

Comment:
I has a smell that cdt-lsp is providing these classes without using them. That's dead code in an exported package. @travkin79 could implement these classes in your custom plug-in where its used?

Reply:
> could implement these classes in your custom plug-in where its used?

Forget my comment, I think it s to be implemented here.

Reply:
These classes are part of the extended `ClangdLanguageServer` interface. Without them, the code will not compile and LSP4J will not be able to parse the LS response. Removing these classes would mean to remove the new methods from the `ClangdLanguageServer` interface, i.e. removing the new clangd LSP extensions.